### PR TITLE
User handle input component

### PR DESCRIPTION
**Pull request by MeBrei**
In the modal for registering the user handle, when choosing the handle, the avatar, handle and availability should be shown in the input field. This PR attempts to create this component, including event handlers which then can be connected to querying the availability. It reuses the `Input/Text` component.

ToDo:

- [x]  error message styling
- [x] ~~remove or move `:focus` color change~~
- [x] move important icon into html
- [x] add success and querying state
- [x] handle key up events
- [x] final style check

Prepares #202

### CODE REVIEW

**Comment by rudolfs** [comment]
Could maybe try with relative/absolute positioning.

Screenshot 2020-03-04 at 10 03 49


```diff
diff --git a/ui/DesignSystem/Component/Avatar.svelte b/ui/DesignSystem/Component/Avatar.svelte
index 1850eed..d501e1d 100644
--- a/ui/DesignSystem/Component/Avatar.svelte
+++ b/ui/DesignSystem/Component/Avatar.svelte
@@ -1,5 +1,5 @@
 <script>
-  import { Title } from "../Primitive";
+  import Title from "../Primitive/Title.svelte";

   export let style = null;

@@ -20,6 +20,12 @@
     border-radius: 17px;
   }

+  .user.small {
+    width: 32px;
+    height: 32px;
+    border-radius: 16px;
+  }
+
   .user.medium {
     width: 36px;
     height: 36px;
diff --git a/ui/DesignSystem/Primitive/Input/Text.svelte b/ui/DesignSystem/Primitive/Input/Text.svelte
index 8f0ad7d..edeca2a 100644
--- a/ui/DesignSystem/Primitive/Input/Text.svelte
+++ b/ui/DesignSystem/Primitive/Input/Text.svelte
@@ -1,5 +1,6 @@
 <script>
   import Text from "../Text.svelte";
+  import Avatar from "../../Component/Avatar.svelte";

   export let style = null;
   export let placeholder = null;
@@ -12,6 +13,12 @@
 </script>

 <style>
+  .wrapper {
+    display: flex;
+    flex-direction: column;
+    position: relative;
+  }
+
   input {
     border: 1px solid var(--color-lightgray);
     padding: 8px;
@@ -19,7 +26,7 @@
     width: 100%;
     height: 48px;
     line-height: 48px;
-    padding: 0 16px 0 16px;
+    padding: 0 16px 0 42px;
     background-color: var(--color-white);
   }
@@ -38,11 +45,11 @@
       url('data:image/svg+xml;utf8,<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2C9.34784 2 6.8043 3.05357 4.92893 4.92893C3.05357 6.8043 2 9.34784 2 12C2 14.6522 3.05357 17.1957 4.92893 19.0711C6.8043 20.9464 9.34784 22 12 22C14.6522 22 17.1957 20.9464 19.0711 19.0711C20.9464 17.1957 22 14.6522 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893ZM10.9393 17.5607C11.2207 17.842 11.6022 18 12 18C12.3978 18 12.7794 17.842 13.0607 17.5607C13.342 17.2794 13.5 16.8978 13.5 16.5C13.5 16.1022 13.342 15.7207 13.0607 15.4394C12.7794 15.158 12.3978 15 12 15C11.6022 15 11.2207 15.158 10.9393 15.4394C10.658 15.7207 10.5 16.1022 10.5 16.5C10.5 16.8978 10.658 17.2794 10.9393 17.5607ZM11 12.1C11.12 13.3 12.87 13.3 13 12.1L13.5 7.10001C13.514 6.96054 13.4986 6.81967 13.4546 6.68656C13.4107 6.55344 13.3393 6.43105 13.245 6.32733C13.1507 6.2236 13.0356 6.14086 12.9073 6.08447C12.7789 6.02809 12.6402 5.99931 12.5 6.00001H11.5C11.3598 5.99931 11.2211 6.02809 11.0927 6.08447C10.9644 6.14086 10.8493 6.2236 10.7551 6.32733C10.6608 6.43105 10.5893 6.55344 10.5454 6.68656C10.5015 6.81967 10.486 6.96054 10.5 7.10001L11 12.1Z" fill="%23E75050"/></svg>')
       no-repeat;
     background-position: right 14px top 55%;
-    padding-right: 46px;
+    padding: 0 46px 0 42px;
   }
 </style>

-<div {style}>
+<div {style} class="wrapper">
   <input
     data-cy={dataCy}
     class:invalid={!valid}
@@ -50,6 +57,7 @@
     bind:value
     {disabled}
     on:change />
+  <Avatar variant="user" size="small" style="width: 32px;height: 48px; justify-content: flex-start;position: absolute; top: 0px; left: 8px" />

   {#if !valid && validationMessage}
     <Text variant="small" style="color: var(--color-red); margin-top: 4px">
diff --git a/ui/Screen/DesignSystemGuide.svelte b/ui/Screen/DesignSystemGuide.svelte
index d46c245..6cbd984 100644
--- a/ui/Screen/DesignSystemGuide.svelte
+++ b/ui/Screen/DesignSystemGuide.svelte
@@ -56,7 +56,6 @@
 </script>

 <style>
-  div,
   table {
     margin-bottom: 32px;
   }
```

**Comment by MeBrei** [comment]
The following is an example integration of the new input variant.
```diff
diff --git a/ui/Screen/RegisterProject/PickNameStep.svelte b/ui/Screen/RegisterProject/PickNameStep.svelte
index 1df55a9..efc3393 100644
--- a/ui/Screen/RegisterProject/PickNameStep.svelte
+++ b/ui/Screen/RegisterProject/PickNameStep.svelte
@@ -38,13 +38,22 @@
   };
 
   let validations = false;
+  let validating = false;
 
-  const validate = () => {
+  const onInput = () => {
+    beginValidation = true;
+    validate();
+  };
+
+  const validate = async () => {
     if (!beginValidation) {
       return;
     }
-
+    validating = true;
     validations = validatejs({ projectName: projectName }, constraints);
+    // timeout for testing to see the pending state
+    await new Promise((r,d) => setTimeout(r,3000));
+    validating = false;
   };
 
   $: validate(projectName);
@@ -55,7 +64,10 @@
   placeholder="Project name"
   bind:value={projectName}
   valid={!(beginValidation && validations && validations.projectName)}
-  validationMessage={beginValidation && validations && validations.projectName && validations.projectName[0]} />
+  validationMessage={beginValidation && validations && validations.projectName && validations.projectName[0]}
+  variant="register"
+  on:input={onInput}
+  validationPending={validating} />
 
 <Flex style="margin-top: 48px;">
   <div slot="left">
```

File: ui/DesignSystem/Primitive/Input/Text.svelte [hunk 1]
```diff
@@ -34,26 +48,54 @@
...
...ll="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M19.0711 4.92893C17.1957 3.05357 14.6522 2 12 2C9.34784 2 6.8043 3.05357 4.92893 4.92893C3.05357 6.8043 2 9.34784 2 12C2 14.6522 3.05357 17.1957 4.92893 19.0711C6.8043 20.9464 9.34784 22 12 22C14.6522 22 17.1957 20.9464 19.0711 19.0711C20.9464 17.1957 22 14.6522 22 12C22 9.34784 20.9464 6.8043 19.0711 4.92893ZM10.9393 17.5607C11.2207 17.842 11.6022 18 12 18C12.3978 18 12.7794 17.842 13.0607 17.5607C13.342 17.2794 13.5 16.8978 13.5 16.5C13.5 16.1022 13.342 15.7207 13.0607 15.4394C12.7794 15.158 12.3978 15 12 15C11.6022 15 11.2207 15.158 10.9393 15.4394C10.658 15.7207 10.5 16.1022 10.5 16.5C10.5 16.8978 10.658 17.2794 10.9393 17.5607ZM11 12.1C11.12 13.3 12.87 13.3 13 12.1L13.5 7.10001C13.514 6.96054 13.4986 6.81967 13.4546 6.68656C13.4107 6.55344 13.3393 6.43105 13.245 6.32733C13.1507 6.2236 13.0356 6.14086 12.9073 6.08447C12.7789 6.02809 12.6402 5.99931 12.5 6.00001H11.5C11.3598 5.99931 11.2211 6.02809 11.0927 6.08447C10.9644 6.14086 10.8493 6.2236 10.7551 6.32733C10.6608 6.43105 10.5893 6.55344 10.5454 6.68656C10.5015 6.81967 10.486 6.96054 10.5 7.10001L11 12.1Z" fill="%23E75050"/></svg>')
```
Comment (rudolfs) [inline ui/DesignSystem/Primitive/Input/Text.svelte]: Phew, I'm glad we can get rid of this duplication!

File: ui/DesignSystem/Primitive/Input/Text.svelte [hunk 2]
```diff
@@ -9,9 +11,17 @@
   export let disabled = null;
   export let valid = true;
   export let validationMessage = null;
+  export let variant = "plain"; // plain || register
```
[the remark this answers is not in the document]
Comment (rudolfs) [reply ui/DesignSystem/Primitive/Input/Text.svelte]: For buttons we use "vanilla" for the basic variant, maybe we can unify the naming?
I also find `register` a bit too generic, how about `handle`, as it is used for user/org handles?

File: ui/DesignSystem/Primitive/Icon/Spinner.svelte [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    0.0785633 13.5845 0.228409 14.3419L2.19038 13.9538C2.06569 13.3235 2 12.6704
+    2 12C2 11.3296 2.06569 10.6765 2.19038 10.0462L0.228409 9.65806ZM2.02122
+    18.6674L3.68315 17.5548C4.41381 18.6461 5.35388 19.5862 6.44525
+    20.3168L5.33261 21.9788C4.024 21.1027 2.89731 19.976 2.02122
+    18.6674ZM9.65806 23.7716L10.0462 21.8096C10.6765 21.9343 11.3296 22 12
+    22C12.6704 22 13.3235 21.9343 13.9538 21.8096L14.3419 23.7716C13.5845
+    23.9214 12.8014 24 12 24C11.1986 24 10.4155 23.9214 9.65806 23.7716ZM18.6674
+    21.9788L17.5548 20.3168C18.6461 19.5862 19.5862 18.6461 20.3168
+    17.5547L21.9788 18.6674C21.1027 19.976 19.976 21.1027 18.6674 21.9788Z" />
+  <animateTransform
```
[the remark this answers is not in the document]
Comment (MeBrei) [reply ui/DesignSystem/Primitive/Icon/Spinner.svelte]: I changed the implementation which improved the performance for me.

**Pull request merged**